cmd/zoekt-sourcegraph-indexserver/sg.go

```diff
@@ -160,3 +172,85 @@ func ping(root *url.URL) error {
 	}
 	return nil
 }
+
+type sourcegraphFake struct {
+	RootDir string
+	Log     *log.Logger
+}
+
+func (sf sourcegraphFake) GetIndexOptions(repos ...string) ([]indexOptionsItem, error) {
+	var items []indexOptionsItem
+	for _, name := range repos {
+		opts, err := sf.getIndexOptions(name)
+		if err != nil {
+			items = append(items, indexOptionsItem{Error: err.Error()})
+		} else {
+			items = append(items, indexOptionsItem{IndexOptions: opts})
+		}
+	}
+	return items, nil
+}
+
+func (sf sourcegraphFake) getIndexOptions(name string) (IndexOptions, error) {
+	dir := filepath.Join(sf.RootDir, filepath.FromSlash(name))
+
+	opts := IndexOptions{
+		RepoID:  int32(crc32.ChecksumIEEE([]byte(name))),
```

Comment:
`% (1<<31)` would prevent negative RepoIDs more directly

Reply:
cool thanks. Also want to ensure it is positive (ie not zero). But I like the inline method. Will do.